### PR TITLE
Test GatedFrameClock properly

### PR DESCRIPTION
**Pull request by jingibus**
Followup for #101

### CODE REVIEW

File: molecule/molecule-runtime/src/commonJvmTest/kotlin/app/cash/molecule/MoleculeTest.kt [hunk 1]
```diff
@@ -264,6 +264,14 @@ class MoleculeTest {
     value = values.awaitValue()
     assertEquals(2, value)
 
+    dispatcher.advanceTimeBy(100)
+    dispatcher.advanceTimeBy(100)
+    dispatcher.advanceTimeBy(100)
```
Comment (JakeWharton) [inline molecule/molecule-runtime/src/commonJvmTest/kotlin/app/cash/molecule/MoleculeTest.kt]: Is this meaningfully different than 300? Or just being explicit?

**Review by JakeWharton — APPROVED** [review]

File: molecule/molecule-runtime/src/commonJvmTest/kotlin/app/cash/molecule/MoleculeTest.kt (hunk 1 above)
Comment (jingibus) [reply molecule/molecule-runtime/src/commonJvmTest/kotlin/app/cash/molecule/MoleculeTest.kt]: The goal is to bump against the backpressure mechanism. So:

1. Write into snapshot state (3), emitter called, bump against the rendezvous channel
2. Write into snapshot state (4), emitter not called
3. Write into snapshot state (5), emitter not called
4. Pull from channel, `moleculeFlow` calls `isRunning = true`, emitter called

So the answer to your question is "I have no idea"

Comment (jingibus) [reply molecule/molecule-runtime/src/commonJvmTest/kotlin/app/cash/molecule/MoleculeTest.kt]: Real answer: Doesn't matter. Switched to one call with `300`, still fails. Revised and pushed

Comment (JakeWharton) [reply molecule/molecule-runtime/src/commonJvmTest/kotlin/app/cash/molecule/MoleculeTest.kt]: Excellent. I fell asleep or I would have merged either way.

**Pull request merged**